atomic_reactor/inner.py

```diff
@@ -434,7 +437,11 @@ def __init__(
         :param plugin_files: load plugins also from these files
         :param client_version: osbs-client version used to render build json
         """
+        self._build_dir = build_dir
+
         self.source = source or DummySource(None, None)
+        # Will be set later to once the platforms are determined for the build.
+        self.build_dirs: Optional[RootBuildDir] = None
```

Comment:
I'm thinking about how to do this for source containers. There is no check_and_set_platforms plugin in the source container workflow, maybe it would be better to pass an already initialized `RootBuildDir` object (or a subclass) to workflow (instead of passing `build_dir: Path`).

The problem of course, is that the prebuild task cannot do that, because it doesn't know what platforms there are yet. Maybe the `RootBuildDir` initialization has to be a 3-step process:
```python
# in the execute method (for source containers, a subclass might be better)
d = RootBuildDir(path)
# binary containers: in the check_and_set_platforms
# source containers: probably never, there are no platforms
d.set_platforms(platforms)
# binary containers: in the check_and_set_platforms plugin
# source containers: in the execute method?
d.copy_sources(source)
```
This doesn't have to  be resolved in this PR though, just throwing out ideas for the next step when we use this with source containers